src/main/java/zmaster587/advancedRocketry/tile/TileGuidanceComputer.java

```diff
@@ -137,41 +147,73 @@ else if(itemType instanceof ItemSatelliteIdentificationChip) {
 	 * @return
 	 */
 	public Vector3F<Float> getLandingLocation(int landingDimension, boolean commit) {
+		//Caution Side-Effect dependency: May require a call to getDestinationDimId to populate correct coordinates.
 		ItemStack stack = getStackInSlot(0);
-		if(!stack.isEmpty() && stack.getItem() instanceof ItemStationChip) {
-			ItemStationChip chip = (ItemStationChip)stack.getItem();
-			if(landingDimension == Configuration.spaceDimId) {
-				//TODO: handle Exception
-				ISpaceObject object = SpaceObjectManager.getSpaceManager().getSpaceStation(ItemStationChip.getUUID(stack));
-				HashedBlockPosition vec = null;
-				if(object instanceof SpaceObject) {
-					if(landingLoc.get(object.getId()) != null) {
-						vec = landingLoc.get(object.getId());
-
-						if(commit)
-							((SpaceObject)object).getPadAtLocation(landingLoc.get(object.getId())).setOccupied(true);
+		//ToDo: replace all nulls with current coordinates of the ship.
```

Comment:
```suggestion
		//TODO: replace all nulls with current coordinates of the ship.
```
minor edit for consistency.